src/components/SigningModals/MultiScreenModal.js

```diff
@@ -0,0 +1,180 @@
+          </h1>
+          {content({
+            prevScreen: prev,
+            nextScreen: next,
+            closeModal: onClose,
+          })}
+        </React.Fragment>
+      )
+    },
+    [prev, next, onClose, screens, smallMode]
+  )
+
+  const onStart = useCallback(() => {
+    // Don’t animate or set the static height when the modal first opens
+    if (firstStart) {
+      setFirstStart(false)
+
+      return
+    }
+
+    setMeasuredHeight(true)
+  }, [firstStart])
+
+  const handleModalClose = useCallback(() => {
+    if (!disableClose) {
+      onClose()
+    }
+  }, [disableClose, onClose])
+
+  return (
+    <Viewport>
+      {({ width: viewportWidth }) => {
+        const gutter = 30
+        const containWidth = viewportWidth < modalWidth + gutter
+
+        return (
+          <Modal
+            padding={0}
+            width={containWidth ? viewportWidth - gutter : modalWidth}
+            onClose={handleModalClose}
+            visible={visible}
+            closeButton={!disableClose}
+          >
+            <Spring
+              config={springs.swift}
+              to={{ height }}
+              immediate={firstStart}
+              native
+            >
+              {({ height }) => (
+                <AnimatedDiv
+                  style={{
+                    position: 'relative',
+                    height: measuredHeight ? height : 'auto',
+                  }}
+                >
+                  <Transition
+                    config={(_, state) =>
+                      state === 'leave' ? springs.instant : springs.smooth
+                    }
+                    items={step}
+                    immediate={firstStart}
+                    from={{
+                      opacity: 0,
+                      transform: `translate3d(${5 * GU * direction}px, 0, 0)`,
+                    }}
+                    enter={{
+                      opacity: 1,
+                      transform: 'translate3d(0, 0, 0)',
+                    }}
+                    leave={{
+                      opacity: 0,
+                      transform: `translate3d(${5 * GU * -direction}px, 0, 0)`,
+                    }}
+                    onRest={(_, status) => {
+                      if (status === 'update') {
+                        setMeasuredHeight(false)
+                      }
+                    }}
+                    onStart={onStart}
+                    native
+                  >
+                    {currentStep => ({ opacity, transform }) => {
+                      // For better performance we avoid reflows between screen changes by matching the screen width with the modal width
+                      const screenWidth =
+                        screens[currentStep].width || DEFAULT_MODAL_WIDTH
+
+                      return (
+                        <AnimatedDiv
+                          ref={elt => {
+                            if (elt) {
+                              setHeight(elt.clientHeight)
```

Comment:
Just wondering, but should we re-evaluate this every time the `currentStep` changes?

Reply:
Yes! so that the modal is aware of the new height for the incoming step. It's doing that here every time a new one is mounted.